File: languages/php/Bloock/Entity/Identity/CredentialSchema.php
<?php

namespace Bloock\Entity\Identity;

/**
 * Represents the schema information for a credential, including its ID and type.
 */
class CredentialSchema
{
    private string $id;
    private string $type;

    /**
     * Constructs an CredentialSchema object with the specified parameters.
     * @param string $id
     * @param string $type
     */
    public function __construct(string $id, string $type)
    {
        $this->id = $id;
        $this->type = $type;
    }

    public static function fromProto(\Bloock\CredentialSchema $res): CredentialSchema
    {
        return new CredentialSchema($res->getId(), $res->getType());
    }

    /**
     * Gets the id of the credential.
     * @return string
     */
    public function getId(): string
    {
        return $this->id;
    }

    /**
     * Gets the type of the credential.
     * @return string
     */
    public function getType(): string
    {
        return $this->type;
    }

    public function toProto(): \Bloock\CredentialSchema
    {
        $p = new \Bloock\CredentialSchema();
        $p->setId($this->id);
        $p->setType($this->type);
        return $p;
    }
}

File: languages/python/bloock/entity/identity/credential_schema.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto


class CredentialSchema:
    """
    Represents the schema information for a credential, including its ID and type.
    """
    def __init__(self, id: str, type: str) -> None:
        """
        Constructs an CredentialSchema object with the specified parameters.
        :type type: object
        :type id: object
        :rtype: object
        """
        self.id = id
        self.type = type

    @staticmethod
    def from_proto(c: proto.CredentialSchema) -> CredentialSchema:
        return CredentialSchema(
            id=c.id,
            type=c.type
        )

    def to_proto(self) -> proto.CredentialSchema:
        return proto.CredentialSchema(
            id=self.id,
            type=self.type
        )
